core/Command/Db/ConvertType.php

```diff
@@ -254,6 +257,16 @@ protected function getToDBConnection(InputInterface $input, OutputInterface $out
 		if ($input->getOption('port')) {
 			$connectionParams['port'] = $input->getOption('port');
 		}
+		if (strpos($input->getOption('hostname'), ':') !== false)  {
+			// Host variable may carry a socket (or port)
+			[$host, $portOrSocket] = explode(':', $input->getOption('hostname'), 2);
+			if (ctype_digit($portOrSocket)) {
+				$connectionParams['port'] = $portOrSocket; // to be consistent with how we handle this elsewhere we accept this too but don't document it
+			} else {
+				$connectionParams['unix_socket'] = $portOrSocket;
+			}
+			$connectionParams['host'] = $host;
+		}
```

Comment:
```suggestion
if (preg_match('/^(.+)(:(\d+|[^:]+))?$/',$input->getOption('hostname'), $matches)) {
    $connectionParams['host'] = $matches[1];
    if (isset($matches[3])) {
        if (is_numeric($matches[3])) {
            $connectionParams['port'] = $matches[3];
        } else {
            $connectionParams['unix_socket'] = $matches[3];
        }
    }
}
```

Suggestion :

 Using regex is most likely better than explode. `/^(.+):(\d+|[^:]+)$/` matches a string that starts with one or more characters `((.+))` followed by a colon `(:)` and then either a numeric value `(\d+)` or one or more non-colon characters `([^:]+)` until the end of the string `($)`.

The regular expression approach allows you to extract the host name and port number or socket path in a single step, which is more concise than using `explode()` and `ctype_digit()`. Additionally, it allows for more flexibility in the format of the option value.

Reply:
I did not copy the comments but they are relevant.